cinema/views.py

```diff
@@ -56,3 +93,66 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            date = parse_date(date)
+            if date:
+                queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            queryset = queryset.filter(movie_id=movie)
+
+        if self.action in "list":
+            return (
+                queryset
+                .select_related()
+                .annotate(
+                    tickets_available=(
+                            F("cinema_hall__rows")
+                            * F("cinema_hall__seats_in_row")
+                            - Count("tickets")
+                    )
+                )
+            )
+        elif self.action in "retrieve":
+            return queryset.select_related()
+        return queryset.distinct()
+
+
+class OrderSetPagination(PageNumberPagination):
+    page_size = 2
+    page_size_query_param = "page_size"
+    max_page_size = 10
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderSetPagination
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(user=self.request.user)
```

Comment:
Ensure that filtering by `user=self.request.user` is secure and that the `request.user` is correctly set, especially if authentication is involved.